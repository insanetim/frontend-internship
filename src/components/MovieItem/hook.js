import { useNavigate } from 'react-router-dom'

const useContainer = id => {
  const navigate = useNavigate()

  const handleClick = () => navigate(`/movie/${id}`)

  return { handleClick }
}

export default useContainer
