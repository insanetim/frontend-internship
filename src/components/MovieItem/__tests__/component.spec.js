import { shallow } from 'enzyme'

import MovieItem from '../component'

const mockedHookData = {
  handleClick: jest.fn()
}
jest.mock('../hook', () => jest.fn(() => mockedHookData))

jest.mock('src/utils', () => ({
  bindId: jest.fn(node => node)
}))

describe('MovieItem component', () => {
  it('matches snapshot', () => {
    const props = {
      movie: {
        id: 1,
        title: 'test/title',
        overview: 'test/overview',
        poster_path: 'test/image'
      },
      actions: [<div key='delete' />]
    }
    const component = shallow(<MovieItem {...props} />)
    expect(component).toMatchSnapshot()
  })

  it('matches snapshot with NoImage', () => {
    const props = {
      movie: {
        id: 1,
        title: 'test/title',
        overview: 'test/overview',
        poster_path: null
      }
    }
    const component = shallow(<MovieItem {...props} />)
    expect(component).toMatchSnapshot()
  })
})
