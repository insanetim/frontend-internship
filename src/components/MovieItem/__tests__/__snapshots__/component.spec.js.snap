// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieItem component matches snapshot 1`] = `
<ForwardRef
  actions={
    [
      <div />,
    ]
  }
  className="top-margin"
  cover={
    <img
      alt="Poster"
      src="https://image.tmdb.org/t/p/w500test/image"
    />
  }
  hoverable={true}
  onClick={[MockFunction]}
>
  <Meta
    description="test/overview"
    title="test/title"
  />
</ForwardRef>
`;

exports[`MovieItem component matches snapshot with NoImage 1`] = `
<ForwardRef
  actions={[]}
  className="top-margin"
  cover={
    <div
      className="ant-card-cover--no-image"
    >
      <img
        alt=""
        src="test-file-stub"
      />
    </div>
  }
  hoverable={true}
  onClick={[MockFunction]}
>
  <Meta
    description="test/overview"
    title="test/title"
  />
</ForwardRef>
`;
